lib/Parse/ParseType.cpp

```diff
@@ -1375,14 +1374,15 @@ bool Parser::isOptionalToken(const Token &T) const {
   // A postfix '?' by itself is obviously optional.
   if (T.is(tok::question_postfix))
     return true;
-  
+
+  if (!T.is(tok::oper_postfix) && !T.is(tok::oper_binary_unspaced)) {
+    return false;
+  }
+
   // A postfix or bound infix operator token that begins with '?' can be
```

Comment:
I think it's clearer if you restore this to the way it was, since the comment applies to the code moved above. While the logic may be the same (I'd have to think about it), it's literally breaking up one test into two separate thoughts.